internal/processor/filtermatcher/attributematcher.go

```diff
@@ -105,8 +105,22 @@ func (ma attributesMatcher) Match(attrs pdata.AttributeMap) bool {
 				return false
 			}
 		} else if property.AttributeValue != nil {
-			if !attr.Equal(*property.AttributeValue) {
+			propertyAttrValue := property.AttributeValue
+			switch propertyAttrValue.Type() {
+			// Case when Attribute Values are of list values.
+			case pdata.AttributeValueARRAY:
+				for i := 0; i < propertyAttrValue.ArrayVal().Len(); i++ {
+					// Check attr value is exists in the AttributeValue array.
+					// Equal checks for any of int, string, bool and double.
+					if attr.Equal(propertyAttrValue.ArrayVal().At(i)) {
+						return true
+					}
+				}
 				return false
+			default:
```

Comment:
Instead of doing this here I suggest that we move this code to `attr.Equal`. It already has a TODO comment that expects this code to be there.

Reply:
That right. I tried doing that earlier.
https://github.com/open-telemetry/opentelemetry-collector/pull/1936#discussion_r503039223